File: remove_duplicates_from_sorted_arr.py
# Tc : O(n)
# sc : O(1)
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 1
        
        j = 1
        counter = 1
        while j <len(nums):
            if nums[j] == nums [j-1]:
                counter = counter + 1
            else:
                counter = 1
            
            if counter <= 2:
                nums[i] = nums[j]
                i += 1
            j += 1
        return i     
            
        
                

File: RemoveDuplicatesFromSortedArr.java
// Time Complexity : O(N)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Three line explanation of solution in plain english

// Your code here along with comments explaining your approach

class Solution {
    public int removeDuplicates(int[] nums) {
        int slow = 0;
        int fast = 1;
        int count = 1;
        while(fast<nums.length){
            if(nums[fast]==nums[fast-1]){
                count++;
            }else{
                count = 1;
            }
            if(count<=2){                
                slow++;
                nums[slow] = nums[fast];
                
            }
            
            fast++;
        }
        return slow+1;
    }
}